app/src/ui/history/commit-list.tsx

```diff
@@ -97,19 +97,23 @@ export class CommitList extends React.Component<ICommitListProps, {}> {
     const tagsToPushSet = new Set(this.props.tagsToPush || [])
 
     const isLocal = this.props.localCommitSHAs.includes(commit.sha)
-    const hasUnpushedTags = commit.tags.some(tagName =>
+    const numUnpushedTags = commit.tags.filter(tagName =>
       tagsToPushSet.has(tagName)
-    )
+    ).length
 
     const showUnpushedIndicator =
-      (isLocal || hasUnpushedTags) && this.props.isLocalRepository === false
+      (isLocal || numUnpushedTags > 0) && this.props.isLocalRepository === false
 
     return (
       <CommitListItem
-        key={commitListItemHash(commit)}
+        key={commit.sha}
```

Comment:
@outofambit I've changed this `key` to just use the commit sha and modified the `shouldComponentUpdate` function on the `CommitListItem` component to take into account them.

I think it's better to just re-render the component when the tags change rather than unmounting/mounting.

Related to that, I think that we should remove the `shouldComponentUpdate` method from the `commitListItem` (it's not obvious that any new property that gets added to `commitListItem` has to be added there) and instead make the `CommitListItem` a `PureComponent` (since we use immutable data structures everywhere this should be fine). I can prepare a separate PR for that.

Reply:
Created https://github.com/desktop/desktop/pull/9672